### ISSUE TITLE
Failed to create cluster with Rook 1.6 and Ceph 15.2.9

### ISSUE THREAD

**Issue by ruruiye** [rook/rook#7731]
I am trying to create a new cluster for bear metal environment for 3 nodes. I use helm chat for the operator, and create a cluster with cluster.yaml. the 3 nodes has 3 disks, and I want to use the /dev/sdc, which is a large disk about 220 TB, the under hold is raid 5. but the cluster failed at prepare OSD job.
 
**Is this a bug report or feature request?**
* Bug Report
```
2021-04-22 12:49:22.077450 I | rookcmd: starting Rook v1.6.0 with arguments '/rook/rook ceph osd provision'
2021-04-22 12:49:22.077583 I | rookcmd: flag values: --cluster-id=f495697d-89af-40ab-af36-6ad0436208da, --cluster-name=rook-ceph, --data-device-filter=sdc, --data-device-path-filter=, --data-devices=, --encrypted-device=false, --force-format=false, --help=false, --location=, --log-flush-frequency=5s, --log-level=DEBUG, --metadata-device=, --node-name=master01.deepmap.io, --operator-image=, --osd-crush-device-class=, --osd-database-size=0, --osd-wal-size=576, --osds-per-device=1, --pvc-backed-osd=false, --service-account=
2021-04-22 12:49:22.077596 I | op-mon: parsing mon endpoints: a=10.43.86.132:6789,b=10.43.23.59:6789,c=10.43.242.199:6789
2021-04-22 12:49:22.114541 I | op-osd: CRUSH location=root=default host=master01-deepmap-io
2021-04-22 12:49:22.114561 I | cephcmd: crush location of osd: root=default host=master01-deepmap-io
2021-04-22 12:49:22.114571 D | exec: Running command: dmsetup version
2021-04-22 12:49:22.165648 I | cephosd: Library version:   1.02.171-RHEL8 (2020-05-28)
Driver version:    4.41.0
2021-04-22 12:49:22.187446 D | cephclient: No ceph configuration override to merge as "rook-config-override" configmap is empty
2021-04-22 12:49:22.187477 I | cephclient: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2021-04-22 12:49:22.187701 I | cephclient: generated admin config in /var/lib/rook/rook-ceph
2021-04-22 12:49:22.187897 D | cephclient: config file @ /etc/ceph/ceph.conf: [global]
fsid                = c9d04d6d-9e75-45f5-9a25-cf186ba6f186
mon initial members = a b c
mon host            = [v2:10.43.86.132:3300,v1:10.43.86.132:6789],[v2:10.43.23.59:3300,v1:10.43.23.59:6789],[v2:10.43.242.199:3300,v1:10.43.242.199:6789]
public addr         = 10.42.0.21
cluster addr        = 10.42.0.21

[client.admin]
keyring = /var/lib/rook/rook-ceph/client.admin.keyring
...
[2021-04-22 12:49:24,972][ceph_volume.util.system][ERROR ] Failed to read contents from: /sys/block/sda/device/sas_address
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ceph_volume/util/system.py", line 91, in get_file_contents
    contents = open_file.read().strip()
OSError: [Errno 19] No such device
[2021-04-22 12:49:24,976][ceph_volume.util.system][ERROR ] Failed to read contents from: /sys/block/sdb/device/sas_address
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ceph_volume/util/system.py", line 91, in get_file_contents
    contents = open_file.read().strip()
OSError: [Errno 19] No such device
[2021-04-22 12:49:24,978][ceph_volume.util.system][ERROR ] Failed to read contents from: /sys/block/sdc/device/sas_address
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ceph_volume/util/system.py", line 91, in get_file_contents
    contents = open_file.read().strip()
OSError: [Errno 19] No such device
[2021-04-22 12:49:24,980][ceph_volume.process][INFO  ] Running command: /usr/sbin/lvs --noheadings --readonly --separator=";" -a --units=b --nosuffix -S lv_path=/dev/sdc -o lv_tags,lv_path,lv_name,vg_name,lv_uuid,lv_size
[2021-04-22 12:49:25,073][ceph_volume.process][INFO  ] stderr Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
[2021-04-22 12:49:25,073][ceph_volume.process][INFO  ] stderr Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
[2021-04-22 12:49:25,074][ceph_volume.process][INFO  ] Running command: /usr/bin/lsblk --nodeps -P -o NAME,KNAME,MAJ:MIN,FSTYPE,MOUNTPOINT,LABEL,UUID,RO,RM,MODEL,SIZE,STATE,OWNER,GROUP,MODE,ALIGNMENT,PHY-SEC,LOG-SEC,ROTA,SCHED,TYPE,DISC-ALN,DISC-GRAN,DISC-MAX,DISC-ZERO,PKNAME,PARTLABEL /dev/sdc
[2021-04-22 12:49:25,085][ceph_volume.process][INFO  ] stdout NAME="sdc" KNAME="sdc" MAJ:MIN="8:32" FSTYPE="" MOUNTPOINT="" LABEL="" UUID="" RO="0" RM="0" MODEL="LOGICAL VOLUME  " SIZE="203.8T" STATE="running" OWNER="root" GROUP="disk" MODE="brw-rw----" ALIGNMENT="0" PHY-SEC="4096" LOG-SEC="512" ROTA="1" SCHED="none" TYPE="disk" DISC-ALN="0" DISC-GRAN="0B" DISC-MAX="0B" DISC-ZERO="0" PKNAME="" PARTLABEL=""
[2021-04-22 12:49:25,086][ceph_volume.process][INFO  ] Running command: /usr/sbin/blkid -p /dev/sdc
[2021-04-22 12:49:25,096][ceph_volume.process][INFO  ] stdout /dev/sdc: PTUUID="c56e3ca8-8f41-4b49-877d-c20ef0f0db50" PTTYPE="gpt"
[2021-04-22 12:49:25,097][ceph_volume.process][INFO  ] Running command: /usr/sbin/pvs --noheadings --readonly --units=b --nosuffix --separator=";" -o vg_name,pv_count,lv_count,vg_attr,vg_extent_count,vg_free_count,vg_extent_size /dev/sdc
[2021-04-22 12:49:25,173][ceph_volume.process][INFO  ] stderr Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
[2021-04-22 12:49:25,173][ceph_volume.process][INFO  ] stderr Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
[2021-04-22 12:49:25,173][ceph_volume.process][INFO  ] stderr Failed to find device for physical volume "/dev/sdc".
[2021-04-22 12:49:25,174][ceph_volume.process][INFO  ] Running command: /usr/bin/ceph-bluestore-tool show-label --dev /dev/sdc
[2021-04-22 12:49:25,208][ceph_volume.process][INFO  ] stderr unable to read label for /dev/sdc: (2) No such file or directory
[2021-04-22 12:49:25,209][ceph_volume.process][INFO  ] Running command: /usr/bin/ceph-bluestore-tool show-label --dev /dev/sdc
[2021-04-22 12:49:25,287][ceph_volume.process][INFO  ] stderr unable to read label for /dev/sdc: (2) No such file or directory
[2021-04-22 12:49:25,288][ceph_volume.process][INFO  ] Running command: /usr/sbin/udevadm info --query=property /dev/sdc
[2021-04-22 12:49:25,301][ceph_volume.process][INFO  ] stdout DEVLINKS=/dev/disk/by-path/pci-0000:5e:00.0-scsi-0:1:0:2 /dev/disk/by-id/wwn-0x600508b1001c65c18ebdac7548b54b2c /dev/disk/by-id/scsi-3600508b1001c65c18ebdac7548b54b2c
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout DEVNAME=/dev/sdc
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout DEVPATH=/devices/pci0000:5d/0000:5d:00.0/0000:5e:00.0/host14/target14:1:0/14:1:0:2/block/sdc
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout DEVTYPE=disk
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_BUS=scsi
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_MODEL=LOGICAL_VOLUME
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_MODEL_ENC=LOGICAL\x20VOLUME\x20\x20
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_PART_TABLE_TYPE=gpt
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_PART_TABLE_UUID=c56e3ca8-8f41-4b49-877d-c20ef0f0db50
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_PATH=pci-0000:5e:00.0-scsi-0:1:0:2
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_PATH_TAG=pci-0000_5e_00_0-scsi-0_1_0_2
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_REVISION=2.93
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_SCSI=1
[2021-04-22 12:49:25,302][ceph_volume.process][INFO  ] stdout ID_SERIAL=3600508b1001c65c18ebdac7548b54b2c
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout ID_SERIAL_SHORT=600508b1001c65c18ebdac7548b54b2c
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout ID_TYPE=disk
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout ID_VENDOR=UN
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout ID_VENDOR_ENC=UN\x20\x20\x20\x20\x20\x20
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout ID_WWN=0x600508b1001c65c1
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout ID_WWN_VENDOR_EXTENSION=0x8ebdac7548b54b2c
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout ID_WWN_WITH_EXTENSION=0x600508b1001c65c18ebdac7548b54b2c
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout MAJOR=8
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout MINOR=32
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout SUBSYSTEM=block
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout TAGS=:systemd:
[2021-04-22 12:49:25,303][ceph_volume.process][INFO  ] stdout USEC_INITIALIZED=5982949
[2021-04-22 12:49:27,271][ceph_volume.main][INFO  ] Running command: ceph-volume  raw prepare --bluestore --data /dev/sdc
[2021-04-22 12:49:27,273][ceph_volume.process][INFO  ] Running command: /usr/bin/lsblk -plno KNAME,NAME,TYPE
[2021-04-22 12:49:27,282][ceph_volume.process][INFO  ] stdout /dev/sda  /dev/sda                          disk
[2021-04-22 12:49:27,283][ceph_volume.process][INFO  ] stdout /dev/sda1 /dev/sda1                         part
[2021-04-22 12:49:27,283][ceph_volume.process][INFO  ] stdout /dev/sda2 /dev/sda2                         part
[2021-04-22 12:49:27,283][ceph_volume.process][INFO  ] stdout /dev/sda3 /dev/sda3                         part
[2021-04-22 12:49:27,283][ceph_volume.process][INFO  ] stdout /dev/sdb  /dev/sdb                          disk
[2021-04-22 12:49:27,283][ceph_volume.process][INFO  ] stdout /dev/sdb1 /dev/sdb1                         part
[2021-04-22 12:49:27,283][ceph_volume.process][INFO  ] stdout /dev/sdc  /dev/sdc                          disk
[2021-04-22 12:49:27,283][ceph_volume.process][INFO  ] stdout /dev/dm-0 /dev/mapper/ubuntu--vg-ubuntu--lv lvm
[2021-04-22 12:49:27,285][ceph_volume.util.system][ERROR ] Failed to read contents from: /sys/block/sda/device/sas_address
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ceph_volume/util/system.py", line 91, in get_file_contents
    contents = open_file.read().strip()
OSError: [Errno 19] No such device
[2021-04-22 12:49:27,289][ceph_volume.util.system][ERROR ] Failed to read contents from: /sys/block/sdb/device/sas_address
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ceph_volume/util/system.py", line 91, in get_file_contents
    contents = open_file.read().strip()
OSError: [Errno 19] No such device
[2021-04-22 12:49:27,291][ceph_volume.util.system][ERROR ] Failed to read contents from: /sys/block/sdc/device/sas_address
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ceph_volume/util/system.py", line 91, in get_file_contents
    contents = open_file.read().strip()
OSError: [Errno 19] No such device
[2021-04-22 12:49:27,293][ceph_volume.process][INFO  ] Running command: /usr/sbin/lvs --noheadings --readonly --separator=";" -a --units=b --nosuffix -S lv_path=/dev/sdc -o lv_tags,lv_path,lv_name,vg_name,lv_uuid,lv_size
[2021-04-22 12:49:27,445][ceph_volume.process][INFO  ] stderr Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
[2021-04-22 12:49:27,445][ceph_volume.process][INFO  ] stderr Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
[2021-04-22 12:49:27,446][ceph_volume.process][INFO  ] Running command: /usr/bin/lsblk --nodeps -P -o NAME,KNAME,MAJ:MIN,FSTYPE,MOUNTPOINT,LABEL,UUID,RO,RM,MODEL,SIZE,STATE,OWNER,GROUP,MODE,ALIGNMENT,PHY-SEC,LOG-SEC,ROTA,SCHED,TYPE,DISC-ALN,DISC-GRAN,DISC-MAX,DISC-ZERO,PKNAME,PARTLABEL /dev/sdc
[2021-04-22 12:49:27,457][ceph_volume.process][INFO  ] stdout NAME="sdc" KNAME="sdc" MAJ:MIN="8:32" FSTYPE="" MOUNTPOINT="" LABEL="" UUID="" RO="0" RM="0" MODEL="LOGICAL VOLUME  " SIZE="203.8T" STATE="running" OWNER="root" GROUP="disk" MODE="brw-rw----" ALIGNMENT="0" PHY-SEC="4096" LOG-SEC="512" ROTA="1" SCHED="none" TYPE="disk" DISC-ALN="0" DISC-GRAN="0B" DISC-MAX="0B" DISC-ZERO="0" PKNAME="" PARTLABEL=""
[2021-04-22 12:49:27,458][ceph_volume.process][INFO  ] Running command: /usr/sbin/blkid -p /dev/sdc
[2021-04-22 12:49:27,469][ceph_volume.process][INFO  ] stdout /dev/sdc: PTUUID="c56e3ca8-8f41-4b49-877d-c20ef0f0db50" PTTYPE="gpt"
[2021-04-22 12:49:27,470][ceph_volume.process][INFO  ] Running command: /usr/sbin/pvs --noheadings --readonly --units=b --nosuffix --separator=";" -o vg_name,pv_count,lv_count,vg_attr,vg_extent_count,vg_free_count,vg_extent_size /dev/sdc
[2021-04-22 12:49:27,560][ceph_volume.process][INFO  ] stderr Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
[2021-04-22 12:49:27,561][ceph_volume.process][INFO  ] stderr Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
[2021-04-22 12:49:27,561][ceph_volume.process][INFO  ] stderr Failed to find device for physical volume "/dev/sdc".
[2021-04-22 12:49:27,562][ceph_volume.process][INFO  ] Running command: /usr/bin/ceph-bluestore-tool show-label --dev /dev/sdc
[2021-04-22 12:49:27,596][ceph_volume.process][INFO  ] stderr unable to read label for /dev/sdc: (2) No such file or directory
[2021-04-22 12:49:27,597][ceph_volume.process][INFO  ] Running command: /usr/bin/ceph-bluestore-tool show-label --dev /dev/sdc
[2021-04-22 12:49:27,689][ceph_volume.process][INFO  ] stderr unable to read label for /dev/sdc: (2) No such file or directory
[2021-04-22 12:49:27,689][ceph_volume.process][INFO  ] Running command: /usr/sbin/udevadm info --query=property /dev/sdc
[2021-04-22 12:49:27,703][ceph_volume.process][INFO  ] stdout DEVLINKS=/dev/disk/by-id/scsi-3600508b1001c65c18ebdac7548b54b2c /dev/disk/by-path/pci-0000:5e:00.0-scsi-0:1:0:2 /dev/disk/by-id/wwn-0x600508b1001c65c18ebdac7548b54b2c
[2021-04-22 12:49:27,703][ceph_volume.process][INFO  ] stdout DEVNAME=/dev/sdc
[2021-04-22 12:49:27,703][ceph_volume.process][INFO  ] stdout DEVPATH=/devices/pci0000:5d/0000:5d:00.0/0000:5e:00.0/host14/target14:1:0/14:1:0:2/block/sdc
[2021-04-22 12:49:27,703][ceph_volume.process][INFO  ] stdout DEVTYPE=disk
[2021-04-22 12:49:27,703][ceph_volume.process][INFO  ] stdout ID_BUS=scsi
[2021-04-22 12:49:27,703][ceph_volume.process][INFO  ] stdout ID_MODEL=LOGICAL_VOLUME
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_MODEL_ENC=LOGICAL\x20VOLUME\x20\x20
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_PART_TABLE_TYPE=gpt
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_PART_TABLE_UUID=c56e3ca8-8f41-4b49-877d-c20ef0f0db50
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_PATH=pci-0000:5e:00.0-scsi-0:1:0:2
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_PATH_TAG=pci-0000_5e_00_0-scsi-0_1_0_2
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_REVISION=2.93
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_SCSI=1
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_SERIAL=3600508b1001c65c18ebdac7548b54b2c
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_SERIAL_SHORT=600508b1001c65c18ebdac7548b54b2c
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_TYPE=disk
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_VENDOR=UN
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_VENDOR_ENC=UN\x20\x20\x20\x20\x20\x20
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_WWN=0x600508b1001c65c1
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_WWN_VENDOR_EXTENSION=0x8ebdac7548b54b2c
[2021-04-22 12:49:27,704][ceph_volume.process][INFO  ] stdout ID_WWN_WITH_EXTENSION=0x600508b1001c65c18ebdac7548b54b2c
[2021-04-22 12:49:27,705][ceph_volume.process][INFO  ] stdout MAJOR=8
[2021-04-22 12:49:27,705][ceph_volume.process][INFO  ] stdout MINOR=32
[2021-04-22 12:49:27,705][ceph_volume.process][INFO  ] stdout SUBSYSTEM=block
[2021-04-22 12:49:27,705][ceph_volume.process][INFO  ] stdout TAGS=:systemd:
[2021-04-22 12:49:27,705][ceph_volume.process][INFO  ] stdout USEC_INITIALIZED=5982949
failed to configure devices: failed to initialize raw based osd: failed to run ceph-volume raw command: exit status 2
```

**Deviation from expected behavior:**
create cluster successfully.

**Expected behavior:**
create cluster successfully.
**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release): ubuntu 18.04.5
* Kernel (e.g. `uname -a`): Linux server00 5.4.0-72-generic #80~18.04.1-Ubuntu SMP Mon Apr 12 23:26:25 UTC 2021 x86_64 x86_64 x86_64 GNU/Linu
* Cloud provider or hardware configuration: 3 nodes with multiple disk, and raid 5 on hardware level
* Rook version (use `rook version` inside of a Rook Pod): v1.6.0
* Storage backend version (e.g. for ceph do `ceph -v`):  15.2.9
* Kubernetes version (use `kubectl version`):  1.20
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): bare metal
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by leseb** [comment]
Looks like a ceph-volume issue, already reported here: https://tracker.ceph.com/issues/43398.
There is nothing we can do in Rook until that tracker gets fixed.

**Comment by ruruiye** [comment]
@leseb  thanks! exactly the problem!

**Comment by szczepl** [comment]
I'm facing exactly the same issue. OSDs cannot be created on old Gen7 machines, 10 worker nodes have /dev/sdb of 600GB available, but no OSDs are created by rook-ceph :(


+-

**Comment by travisn** [comment]
@szczepl You see the same error in the osd prepare log? What release of rook and ceph are you running? Perhaps [this topic](https://rook.io/docs/rook/latest/Troubleshooting/ceph-common-issues/#osd-pods-are-not-created-on-my-devices) would help.

**Comment by szczepl** [comment]
Actually, my colleague found the RC. OSDs are not created, because our domain name in the Openshift cluster and hostnames are very long. Together, with full cluster resource name they extend 63 chars.

The following error is thrown in the rook-ceph namepace job:

  Warning  FailedCreate  3m23s  job-controller  Error creating: Pod "rook-ceph-osd-prepare-xx-xxxx-xxx02-xxx01.xxxxxxxx02.--1-25hr8" is invalid: [metadata.generateName: Invalid value: "rook-ceph-osd-prepare-xx-xxxx-xxx02-xxx01.xxxxxxxx02.--1-": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), metadata.name: Invalid value: "rook-ceph-osd-prepare-xx-xxxx-xxx02-xxx01.xxxxxxxx02.--1-25hr8": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]

For those hostnames which are a bit shorter, this job suceeds, but then OSD pod cannot be created for the same reason.

We will reinstall whole cluster with changing the domain and hostnames to have less characters.

**Comment by travisn** [comment]
Hmm I thought Rook handled longer host names. If you find it repros on a more recent release, would you mind opening a new issue for this? I don't recall a fix for long host names since v1.5, but perhaps I'm forgetting.
